### PR TITLE
[MRESOLVER-433] Config for expect/continue inhibition (#363)

### PR DESCRIPTION
**Pull request by cstamas**
Add configuration to inhibit expect/continue handshake.

---

https://issues.apache.org/jira/browse/MRESOLVER-433
Backport of relevants parts from this commit: https://github.com/apache/maven-resolver/commit/178cfba9f3889f7e942a6a0d74716355b01a78f5

### CODE REVIEW

File: maven-resolver-api/src/main/java/org/eclipse/aether/ConfigurationProperties.java [hunk 1]
```diff
@@ -259,6 +259,60 @@ public final class ConfigurationProperties {
      */
     public static final int DEFAULT_HTTP_MAX_CONNECTIONS_PER_ROUTE = 50;
 
+    /**
+     * The local address (interface) to use with HTTP transport. Not all transport supports this option.
+     *
+     * @since 2.0.0
```
Comment (slawekjaranowski) [inline maven-resolver-api/src/main/java/org/eclipse/aether/ConfigurationProperties.java:265]: 1.9.17 ?

Comment (cstamas) [reply maven-resolver-api/src/main/java/org/eclipse/aether/ConfigurationProperties.java:265]: yes, cherry-pick brought this over...

**Review by slawekjaranowski — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #1102

**Comment by jira-importer** [comment]
Resolve #1106

**Comment by jira-importer** [comment]
Resolve #1102

**Comment by jira-importer** [comment]
Resolve #1106

**Comment by jira-importer** [comment]
Resolve #1106

**Pull request merged**